src/networkaddress.rs

```diff
@@ -107,28 +126,27 @@ impl MulticastJoinable for SocketAddrV4 {
         interface: InterfaceName,
         _token: PrivateToken,
     ) -> std::io::Result<()> {
+        let mode = crate::interface::LinuxNetworkMode::Ipv4;
+
         let request = libc::ip_mreqn {
             imr_multiaddr: libc::in_addr {
                 s_addr: u32::from_ne_bytes(self.ip().octets()),
             },
             imr_address: libc::in_addr {
                 s_addr: u32::from_ne_bytes(Ipv4Addr::UNSPECIFIED.octets()),
             },
-            imr_ifindex: InterfaceDescriptor {
-                interface_name: Some(interface),
-                mode: crate::interface::LinuxNetworkMode::Ipv4,
-            }
-            .get_index()
-            .ok_or(std::io::ErrorKind::InvalidInput)? as _,
```

Comment:
this formats in a (to me) confusing way where it looks like you assign a `InterfaceDescriptor` literal to the `imr_ifindex` field. But no, there are some sneaky method calls that change the whole meaning. This should just be restructured in any case. (I know we want to change this anyway, but still)

Reply:
This is indeed somewhat convoluted, but actually the interfacedescriptor assignment is really close to the intended meaning here, namely that we want it to use the provided interface. The function calls really are to deal with the fact that a conversion is needed.